AWSS3UnitTests/AWSS3PreSignedURLBuilderUnitTests.swift

```diff
@@ -0,0 +1,163 @@
+        return request
+    }
+
+    /// Given: a `AWSS3GetPreSignedURLRequest` with `preferredAccessStyle` set to `.path`
+    /// When: `AWSS3PreSignedURLBuilder.getPresignedURL(_:)` is invoked
+    /// Then: The generated request uses path style to create the URL.
+    func testPreferredAccessStyle_withPathStyle() {
+        let request = createRequest(
+            bucket: "pathbucket",
+            key: "PathKey",
+            preferredAccessStyle: .path
+        )
+
+        let expectation = expectation(description: "getPresignedURL")
+        builder.getPreSignedURL(request).continueWith { task in
+            defer {
+                expectation.fulfill()
+            }
+
+            XCTAssertNil(task.error)
+            guard let presignedURL = task.result as? URL else {
+                XCTFail("Expected result")
+                return
+            }
+
+            guard let components = URLComponents(url: presignedURL, resolvingAgainstBaseURL: true) else {
+                XCTFail("Unable to retrieve components")
+                return
+            }
+
+            XCTAssertEqual(components.host, "s3.us-east-1.amazonaws.com")
+            XCTAssertEqual(components.path, "/pathbucket/PathKey")
+            return nil
+        }
+        waitForExpectations(timeout: 1)
+    }
+
+    /// Given: a `AWSS3GetPreSignedURLRequest` with `preferredAccessStyle` set to `.virtualHosted` and a `bucketName` that is compliant.
+    /// When: `AWSS3PreSignedURLBuilder.getPresignedURL(_:)` is invoked
+    /// Then: The generated request uses Virtual-Hosted style to create the URL.
+    func testPreferredAccessStyle_withVirtualHostedStyle() {
+        let request = createRequest(
+            bucket: "virtualhostedbucket",
+            key: "virtualHostedKey",
+            preferredAccessStyle: .virtualHosted
+        )
+
+        let expectation = expectation(description: "getPresignedURL")
+        builder.getPreSignedURL(request).continueWith { task in
+            defer {
+                expectation.fulfill()
+            }
+
+            XCTAssertNil(task.error)
+            guard let presignedURL = task.result as? URL else {
+                XCTFail("Expected result")
+                return
+            }
+
+            guard let components = URLComponents(url: presignedURL, resolvingAgainstBaseURL: true) else {
+                XCTFail("Unable to retrieve components")
+                return
+            }
+
+            XCTAssertEqual(components.host, "virtualhostedbucket.s3.us-east-1.amazonaws.com")
+            XCTAssertEqual(components.path, "/virtualHostedKey")
+            return nil
+        }
+        waitForExpectations(timeout: 1)
+    }
+
+    /// Given: a `AWSS3GetPreSignedURLRequest` with `preferredAccessStyle` set to `.virtualHosted` and a `bucketName` that is not compliant.
+    /// When: `AWSS3PreSignedURLBuilder.getPresignedURL(_:)` is invoked
+    /// Then: The generated request uses Virtual-Hosted style to create the URL.
```

Comment:
i think this should be `The generated request uses path style to create the URL.` ?